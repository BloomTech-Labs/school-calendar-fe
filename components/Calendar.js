import React, { useState } from 'react';
import dayjs from 'dayjs';
import { Heading, Flex, Grid, Text, Box, Icon, Image } from '@chakra-ui/core';

const Cell = ({
  children,
  fontSize = 'lg',
  height = ['60px', '150px'],
  ...props
}) => (
  <Box
    fontSize={fontSize}
    fontWeight={300}
    textAlign="center"
    h={height}
    {...props}
  >
    {children}
  </Box>
);

const Calendar = ({ events, profile }) => {
  const currentDay = dayjs();
  const [date, setDate] = useState(dayjs());

  const currentYear = date.year();
  const currentMonth = date.month(); // January = 0
  const daysInMonth = date.daysInMonth();

  const firstDayOfMonth = dayjs(`${currentYear}-${currentMonth + 1}-1`);
  const weekDayOfFirstDay = firstDayOfMonth.day(); // Sunday = 0

  const lastDayOfMonth = dayjs(
    `${currentYear}-${currentMonth + 1}-${daysInMonth}`
  );
  const weekDayOfLastDay = lastDayOfMonth.day();

  const weekDays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

  const handlePrev = () => {
    setDate(date.subtract(1, 'month'));
  };

  const handleNext = () => {
    setDate(date.add(1, 'month'));
  };

  return (
    <div>
      <img src={profile && profile.photoUrl} alt="" width="90" />
      <div>
        <h6>{profile && profile.name}</h6>
        <span>{profile && profile.email}</span>
        <br />
      </div>
      <div className="calendar">
        <div className="header">
          <button type="button" className="nav prev" onClick={handlePrev}>
            &lt;
          </button>
          <h3 className="heading">{date.format('MMM DD YYYY')}</h3>
          <button type="button" className="nav nav next" onClick={handleNext}>
            &gt;
          </button>
        </div>
        <Grid
          templateColumns="repeat(7, 1fr)"
          textAlign="right"
          gap={1}
          borderBottomWidth="1px"
          borderBottomColor="gray.200"
        >
          {weekDays.map(d => (
            <Cell fontSize="2xl" height="auto" key={d}>
              {d}
            </Cell>
          ))}
        </Grid>
        <Grid templateColumns="repeat(7, 1fr)" textAlign="right" gap={2}>
          {[...Array(weekDayOfFirstDay).keys()].map(i => (
            <Cell borderBottomWidth="1px" borderBottomColor="gray.200" key={i}>
              <Box backgroundColor="gray.100" h="100%" />
            </Cell>
          ))}
          {[...Array(daysInMonth).keys()].map(i => {
            const day = i + 1;
            const isToday =
              day === currentDay.date() &&
              currentMonth === currentDay.month() &&
              currentYear === currentDay.year();
            const style = {
              color: isToday ? 'indianred' : 'inherit'
            };
            return (
              <Cell
                borderBottomWidth="1px"
                borderBottomColor="gray.200"
                key={i}
              >
                <Flex
                  direction="column"
                  align="center"
                  justify="space-between"
                  h="100%"
                  py={[1, 8]}
                >
                  {day}
                  {events &&
                    events.map(e => {
                      {
                        const event =
                          day === dayjs(e && e.start.dateTime).date() &&
                          currentMonth ===
                            dayjs(e && e.start.dateTime).month() &&
                          currentYear ===
                            dayjs(e && e.start.dateTime).year() ? (
                            <Icon name="circle" color="green.200" key={e.id} /> // e.summary
                          ) : null;
                        return event;
                      }
                    })}
                </Flex>
              </Cell>
            );
          })}
          {[...Array(6 - weekDayOfLastDay).keys()].map(i => (
            <Cell borderBottomWidth="1px" borderBottomColor="gray.200" key={i}>
              <Box backgroundColor="gray.100" h="100%" />
            </Cell>
          ))}
        </Grid>
      </div>
    </div>
  );
};
export default Calendar;
