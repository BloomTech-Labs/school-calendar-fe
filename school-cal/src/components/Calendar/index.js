import React, { useContext, useEffect, useState } from "react"
import { CalendarContext } from "../../contexts/calendar/calendarState"
import { Button, Grid } from "@material-ui/core"
import { makeStyles } from "@material-ui/core/styles"
import AddIcon from "@material-ui/icons/Add"

import CreateEvent from "../Events/CreateEvent"
import EditEvent from "../Events/EditEvent"
import AddSubscribers from "../Events/addSubscriber"
import CustomNotification from "../Events/CustomNotification"
import moment from "moment"

//fullcalendar
import FullCalendar from "@fullcalendar/react"
import dayGridPlugin from "@fullcalendar/daygrid"
import timeGridPlugin from "@fullcalendar/timegrid"
import interactionPlugin from "@fullcalendar/interaction"
import rrulePlugin from "@fullcalendar/rrule"
import "@fullcalendar/core/main.css"
import "@fullcalendar/daygrid/main.css"
import "@fullcalendar/timegrid/main.css"

const useStyles = makeStyles(theme => ({
  headerContainer: {
    marginBottom: theme.spacing(3)
  },
  createButton: {
    backgroundColor: "#F5945B",
    marginRight: theme.spacing(3)
  },
  buttonLabel: {
    textTransform: "none"
  }
}))

const Calendar = props => {
  const classes = useStyles()
  const {
    setUserCalendarEvent,
    userCalendars,
    editUserCalendarEvent
  } = useContext(CalendarContext)

  const [createEvent, openCreateEvent] = useState(false)
  const [isAddSubscriberOpen, setAddSubscribers] = useState(false)
  const [isCustomNotificationOpen, setCustomNotification] = useState(false)
  const [editEvent, openEditEvent] = useState(false)

  const [userCalendarEvents, setUserCalendarEvents] = useState([{ events: [] }])

  const initialCreateEventProperty = {
    startTime: moment()
      .hours(6)
      .minutes(0)
      .seconds(0),
    endTime: moment()
      .hours(7)
      .minutes(0)
      .seconds(0),
    eventTitle: "",
    eventLocation: "",
    eventNote: "",
    isAllDayEvent: false,
    repeatSelection: { id: 1, name: "Never" }
  }
  useEffect(() => {
    if (userCalendars.length > 0) {
      const formatted = userCalendars.map(calendar => {
        if (calendar.showEvents && calendar.events.length > 0) {
          return {
            events: calendar.events.map(event => {
              return {
                id: event.uuid,
                start: event.isAllDayEvent ? event.startDate : event.startTime,
                end: event.isAllDayEvent ? event.endDate : event.endTime,
                title: event.eventTitle,
                location: event.eventLocation,
                note: event.eventNote,
                allDay: event.isAllDayEvent,
                rrule: event.rrule,
                isRepeatingEvent: event.isRepeatingEvent,
                calendarUuid: calendar.uuid,

                duration: Boolean(event.rrule)
                  ? Boolean(event.isAllDayEvent)
                    ? moment
                        .duration(moment(event.endDate).diff(event.startDate))
                        .asMilliseconds()
                    : moment
                        .duration(
                          moment(event.endTime).diff(moment(event.startTime))
                        )
                        .asMilliseconds()
                  : 0
              }
            }),
            color: calendar.calendarColor
          }
        } else {
          return { events: [] }
        }
      })
      setUserCalendarEvents(formatted)
    } else {
      setUserCalendarEvents([{ events: [] }])
    }
  }, [userCalendars])

  // when a user clicks on am event, FullCalendar will invokes this function to initiate the selected event

  const handleEventClick = info => {
    const { id, start, end, title, allDay, extendedProps, _def } = info.event
    console.log("Info ", info.event)
    setUserCalendarEvent({
      startDate: moment(start).format(),
      endDate: allDay
        ? moment(end)
            .subtract(1, "days")
            .format()
        : moment(end).format(),

      startTime: allDay
        ? moment(start)
            .hours(moment().hours())
            .toISOString()
        : moment(start).toISOString(true),
      endTime: allDay
        ? moment(start)
            .hours(
              moment()
                .add(1, "hours")
                .hours()
            )
            .toISOString()
        : extendedProps.isRepeatingEvent
        ? moment(extendedProps.end).toISOString()
        : moment(end).toISOString(true),
      eventTitle: title,
      eventLocation: extendedProps.location,
      eventNote: extendedProps.note,
      isRepeatingEvent: extendedProps.isRepeatingEvent,
      isAllDayEvent: allDay,
      calendarUuid: extendedProps.calendarUuid,
      existingRrule: extendedProps.isRepeatingEvent
        ? _def.recurringDef.typeData.options
        : null,
      uuid: id
    })

    openEditEvent(true)
  }

  const handleDateClick = info => {
    const primaryCalendar = userCalendars.find(calendar => calendar.isDefault)

    setUserCalendarEvent({
      calendarUuid: primaryCalendar.uuid,
      startDate: moment(info.date).format(),
      endDate: moment(info.date).format(),
      startTime: moment(info.date)
        .hours(moment().hours())
        .toISOString(true),
      endTime: moment(info.date)
        .hours(
          moment()
            .add(1, "hours")
            .hours()
        )
        .toISOString(true),
      eventTitle: "",
      eventLocation: "",
      eventNote: "",
      isAllDayEvent: false,
      isRepeatingEvent: false,
      rrule: null
    })
    openCreateEvent(true)
  }

  const handleDatesSelection = info => {
    const primaryCalendar = userCalendars.find(calendar => calendar.isDefault)
    setUserCalendarEvent({
      calendarUuid: primaryCalendar.uuid,
      startDate: moment(info.startStr).format(),
      endDate: moment(info.endStr)
        .subtract(1, "days")
        .format(),

      eventTitle: "",
      eventLocation: "",
      eventNote: "",
      isAllDayEvent: false,
      isRepeatingEvent: false,
      rrule: null
    })
    openCreateEvent(true)
  }

  const handleClosingCreateEvent = () => {
    setUserCalendarEvent(initialCreateEventProperty)
    openCreateEvent(false)
  }

  const handleEventTimeEdit = edit => {
    const { event } = edit
    const changes = {
      startDate: moment(event.start).format("YYYY-MM-DD"),
      endDate: moment(event.end).format("YYYY-MM-DD"),
      startTime: event.allDay
        ? null
        : moment(event.start)
            .hours(moment(event.start).hour())
            .minutes(moment(event.start).minute())
            .seconds(moment(event.start).second())
            .toISOString(true),
      endTime: event.allDay
        ? null
        : moment(event.end)
            .hours(moment(event.end).hour())
            .minutes(moment(event.end).minute())
            .seconds(moment(event.end).second())
            .toISOString(true)
    }
    editUserCalendarEvent(event.id, changes)
  }

  return (
    <div>
      <Grid container className={classes.headerContainer}>
        <Grid item xs={12}>
          <Button
            classes={{
              root: classes.createButton,
              label: classes.buttonLabel
            }}
            startIcon={<AddIcon />}
            onClick={() => openCreateEvent(true)}>
            Create Event
          </Button>
          <Button
            classes={{ root: classes.createButton, label: classes.buttonLabel }}
            startIcon={<AddIcon />}
            onClick={() => setAddSubscribers(true)}>
            Add Subscriber
          </Button>
          <Button
            classes={{ root: classes.createButton, label: classes.buttonLabel }}
            startIcon={<AddIcon />}
            onClick={() => setCustomNotification(true)}>
            Send Notification
          </Button>
        </Grid>
      </Grid>
      <FullCalendar
        header={{
          left: "prev,next today",
          center: "title",
          right: "dayGridMonth,dayGridWeek,dayGridDay"
        }}
        defaultView="dayGridMonth"
        plugins={[
          dayGridPlugin,
          timeGridPlugin,
          interactionPlugin,
          rrulePlugin
        ]}
        eventClick={handleEventClick}
        eventSources={[...userCalendarEvents]}
        dateClick={handleDateClick}
        selectable={true}
        droppable={true}
        editable={true}
        select={handleDatesSelection}
        eventAllow={(dropInfo, draggedEvent) => {
          if (!draggedEvent.extendedProps.isRepeatingEvent) {
            return true
          } else {
            return false
          }
        }}
        eventDrop={handleEventTimeEdit}
      />
      <CreateEvent open={createEvent} handleClose={handleClosingCreateEvent} />
      <EditEvent open={editEvent} handleClose={() => openEditEvent(false)} />
      <AddSubscribers
        open={isAddSubscriberOpen}
        handleClose={() => setAddSubscribers(false)}
      />
       <CustomNotification
        open={isCustomNotificationOpen}
        handleClose={() => setCustomNotification(false)}
      />
    </div>
  )
}

export default Calendar
