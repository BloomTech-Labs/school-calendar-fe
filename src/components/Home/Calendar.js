import React, {useState} from 'react'
import dateFns from 'date-fns'
import { findRenderedDOMComponentWithClass } from 'react-dom/test-utils'

function Calendar() {
    const [ cal , setCal] = useState({
        currentMonth: new Date(),
        selectedDate: new Date ()
    })

    renderHeader() {
        const dateFormat = 'MMMM YYYY';

        return(
            <div className="header row flex-middle">
      <div className="col col-start">
        <div className="icon" onClick={this.prevMonth}>
          chevron_left
        </div>
      </div>
      <div className="col col-center">
        <span>
          {dateFns.format(this.state.currentMonth, dateFormat)}
        </span>
      </div>
      <div className="col col-end" onClick={this.nextMonth}>
        <div className="icon">chevron_right</div>
      </div>
    </div>
        );
    }

    renderDays() {}

    renderCells() {}

    onDateClick = day => {}

    nextMonth = () => {}

    prevMonth = () => {

    }
    return (
        <div className="calendar">
            {renderHeader()}
            {renderDays()}
            {renderCells()}
        </div>
    )
}

export default Calendar
