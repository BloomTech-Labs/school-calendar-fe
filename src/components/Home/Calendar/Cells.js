import React, { useState, useEffect } from "react";
import axios from 'axios'
import {
  format,
  startOfMonth,
  endOfMonth,
  startOfWeek,
  endOfWeek,
  isSameMonth,
  isSameDay,
  parse,
  addDays
} from "date-fns";

const Cells = props => {
  const [data, setData] = useState({events:[{}]});
  const [selectedDate, setSelectedDate] = useState(new Date());
  const [modal, setModal] = useState(false);
  const toggle = () => setModal(!modal);
  const monthStart = startOfMonth(props.currDate);
  const monthEnd = endOfMonth(monthStart);
  const startDate = startOfWeek(monthStart);
  const endDate = endOfWeek(monthEnd);
  const dateFormat = "d";
  const rows = [];
  let days = [];
  let day = startDate;
  let formattedDate = "";

  useEffect(() => {
    (async () => {
      const res = await axios.get("/api/events");
      const results = await res.data;
      localStorage.setItem("googleId:", res.data.googleId)
      console.log("results: ", results);
      setData(results);
    })();
  }, [setData]);
  console.log("andrew data:",data)
  const onDateClick = day => {
    setSelectedDate(day);
  };
  while (day <= endDate) {
    for (let i = 0; i < 7; i++) {
      formattedDate = format(day, dateFormat);
      let eventDate = format(day, 'yyyymmdd')
      const cloneDay = day;
      let eventDescription = data.events[i]
      console.log('single events',data.events[i])
      days.push(
        <div
          className={`column cell ${
            !isSameMonth(day, monthStart)
              ? "disabled"
              : isSameDay(day, selectedDate)
              ? "selected"
              : ""
          }`}
          key={day}
          onClick={() => {
            onDateClick(parse("", "", cloneDay));
            toggle();
          }}
        >
        {}
          
          <span className="number">{formattedDate}</span>
          <span className="bg">{formattedDate}</span>
        </div>
      );
      day = addDays(day, 1);
    }
    rows.push(
      <div className="row" key={day}>
        {" "}
        {days} 
      </div>
    );
    days = [];
  }
  return <div className="body">{rows}
  {/* <ul>
          {data &&
            data.events.map(event => (
              <li key={event.id}>
                <p>
                  {event.start.dateTime}
                  <strong> - {event.summary}</strong>
                </p>
              </li>
            ))}
        </ul> */}
  </div>;
};
export default Cells;