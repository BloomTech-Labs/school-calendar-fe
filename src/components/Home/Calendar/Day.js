import React from 'react'
import dayjs from 'dayjs';
import { useTemplate } from '../../../hooks/useTemplate'


const Day = ({num, events, templateFormOpen, month, year}) => {
  const {selected, setSelected} = useTemplate()
  const {date, setDate} = useTemplate()
  const currentDay = dayjs();

  const isToday = num === currentDay.date()
    && month === currentDay.month() 
    && year === currentDay.year();
  const isPicked = selected.includes(`${year}-${month+1 < 10 ? 0 : ''}${month+1}-${num < 10 ? 0 : ''}${num}`);
  const style = {
    color: isToday ? 'indianred' : 'inherit',
    background: isPicked ?'green': null
  };
  const handleSelected = i => {
    //dateTime: "2020-02-28T08:30:00-08:00"

    //concatinated to w/ turnary to put into correct format
    const newdate = date.format('YYYY-MM').concat(`-${num < 10 ? 0 : ''}${num}`)

    templateFormOpen
    ?(selected.includes(newdate)
      ? setSelected(selected.filter(day => day !== newdate))
      : setSelected(selected.concat(newdate)))
    :alert('pick a template')
    console.log(selected)
  }

  return (
    <span style={style} onClick={()=> handleSelected(num)}>
      {events && events.map(e => {
        const event =
          num === dayjs(e && e.start.dateTime).date() &&
          month === dayjs(e && e.start.dateTime).month() &&
          year === dayjs(e && e.start.dateTime).year()
            ? e.summary
            : null;
        return event;
      })}

      {/* {event && event.start.dateTime} */}
      {num}
    </span>
  )
}

export default Day
