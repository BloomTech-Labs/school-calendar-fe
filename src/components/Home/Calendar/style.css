label {
    display: inline-block;
    padding-right: 0.25rem;
    margin-bottom: 0.5rem;
    text-transform: uppercase;
    letter-spacing: 0.6px;
    text-align: right;
    font-weight: 700;
    font-size: 12px;
    width: 100px;
  }
  
  .labels,
  .dates {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    text-align: right;
    grid-gap: 4px;
  }
  
  .dates span {
    height: 50px;
    font-size: 16px;
    letter-spacing: -1px;
    border: 1px solid #e6e4e4;
    padding-right: 4px;
    font-weight: 700;
  }
  
  .labels span {
    font-weight: 300;
    text-align: center;
    margin-bottom: 0.5rem;
    opacity: 0.6;
  }
  
  .faded {
    opacity: 0.4;
  }
  
  .today {
    background-color: #eee;
    color: indianred;
  }
  