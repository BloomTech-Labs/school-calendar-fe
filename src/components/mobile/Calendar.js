import React, { useState, useEffect, useRef } from 'react';

import dayjs from 'dayjs';
import { Heading, Flex, Grid, Box, IconButton } from '@chakra-ui/core';
import Days from '../dashboardComponents/calendarComponents/Days';
import Cell from '../dashboardComponents/calendarComponents/Cell';
import useDate from '../../hooks/useDate';
import styled from 'styled-components';

//for some reason, a regular useEffect doesn't work but putting this out here like this does
const useMountEffect = (fun, numOfMonths) => useEffect(fun, [numOfMonths])

const Calendar = ({ events, selected, setSelected, templateFormOpen, month,  monthList, setMonths, numOfMonths, setNumOfMonths, i, reloadMonths, setReloadMonths, setStartMonth, startMonth, visibleMonths, setVisibleMonths}) => {

  const currentDay = dayjs();
  // state to display cuurent date
  const [date, setDate] = useState(dayjs());

  //define a month when user scrolls to it will trigger load of additional months (in this case somewhere in the middle of the year)
  const [reloadMonth, setReloadMonth] = useState(monthList[monthList.length-6].format('MMMM'))



  const {
    currentMonth,
    currentYear,
    daysInMonth,
    weekDayOfFirstDoM,
    weekDayOfLastDoM,
    weekDays
  } = useDate(date);

 
  const scrollToRef = (ref) => {
    // window.addEventListener('scroll', ()=>{

       if(ref !==null){
         //needs a setTimeout because ref and virtual dom don't always agree if ref exists or not
        setTimeout(()=>{
          
        let reloadDay = monthList[monthList.length-6].$d
        console.log('dimensions', ref.current.getBoundingClientRect().height)
        
        //if the user scrolls past a month whose name matches the reloadMonth, load more months
        
        //grabs the year of the current ref from it's text and converts to a number for comparison
        let yearIndex = ref.current.innerText.match(/[0-9][0-9][0-9]|[0-9]/).index;
        let yearText = ref.current.innerText.substring(yearIndex, yearIndex+4);
        let yearNum = Number(yearText)
        console.log('scrollY', window.scrollY, 'ref.current', ref.current.offsetTop)
        //checks what month we've scrolled past and if the month and year names match our reloadMonth
        if (window.scrollY > ref.current.offsetTop && ref.current.innerText.substring(0,3) === reloadMonth.substring(0,3) && yearNum === month.$y){
            //if yes, load more months
            setNumOfMonths(numOfMonths + 12)
            
            //check dimensions of calendars above and subtract when changing start and end


            // setNumOfMonths(numOfMonths + 12)
            // setStartMonth(startMonth + 12)

            // window.scrollTo(0, 0);






            // console.log('num of months before', numOfMonths)
            // if(!reloadMonths.includes(reloadDay)){
            //   setNumOfMonths(numOfMonths + 12)
            //   setReloadMonths([...reloadMonths, reloadDay])
            //   // setVisibleMonths({start: numOfMonths, end: numOfMonths+12})
            // }
                // setStartMonth(startMonth + 12)
              

        }

        }, 350)
       } else {
         console.log("LOADING")
       }
    // })
    // return window.removeEventListener('scroll', ()=>console.log('removed'))
    }
  
  

  useEffect(()=>{
    setDate(month)
  },[monthList])

  const myRef = useRef("")
  useMountEffect(() => scrollToRef(myRef),[numOfMonths])
  
  

  // if(i >= visibleMonths.start && i < visibleMonths.end){
    return (
      <Box ref={myRef} id={i} className="calendar" backgroundColor="white" borderRadius="10px">
        <Flex className="header" align="center" justify="center" py={4}>
          <MonthNameContainer>
            {/* <Heading className="heading">{date.format('MMMM')} {date.format('YYYY')}</Heading> */}
            <Heading className="heading">{month.format('MMMM')} {date.format('YYYY')}</Heading>
          </MonthNameContainer>
        </Flex>
        <Grid
          className="weekdays-grid"
          templateColumns="repeat(7, 1fr)"
          textAlign="right"
          borderBottomWidth="1px"
          borderBottomColor="gray.200"
        >
          {weekDays.map(d => (
            <Cell
              className="weekdays-item"
              fontSize={['lg', '3xl']}
              height="auto"
              key={d}
            >
              {d}
            </Cell>
          ))}
        </Grid>
        <Grid
          className="calendar-days-grid"
          templateColumns="repeat(7, 1fr)"
          textAlign="right"
        >
          <Days
            events={events}
            date={date}
            templateFormOpen={templateFormOpen}
            selected={selected}
            setSelected={setSelected}
            weekDayOfFirstDay={weekDayOfFirstDoM}
            weekDayOfLastDay={weekDayOfLastDoM}
            daysInMonth={daysInMonth}
            currentDay={currentDay}
            currentMonth={currentMonth}
            currentYear={currentYear}
          />
        </Grid>
      </Box>
    );
  // } else {
  //   return(
  //     <div style={{height: "817px"}} ref={myRef}>
  //        <Heading className="heading">{month.format('MMMM')} {date.format('YYYY')}</Heading>
  //     </div>
  //   )
  // }
};

export default Calendar;



const MonthNameContainer = styled.div`

  // background: red;
  width: 12%;
  text-align: center;
  box-sizing: border-box;

  @media(max-width: 1700px){
    width: 30%;
  }
  @media(max-width: 1400px){
    width: 25%;
  }
  @media(max-width: 1120px){
    width: 35%;
  }
  @media(max-width: 860px){
    width: 40%;
  }
`;