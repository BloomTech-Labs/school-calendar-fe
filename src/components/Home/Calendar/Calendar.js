import React, { useState } from 'react';

import dayjs from 'dayjs';

// import Loading from '../loadingScreen/loading.js';

import './style.css';

const weekDays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
const currentDay = dayjs();

const Day = ({num, selected, events, setSelected, templateFormOpen}) => {
  const [date, setDate] = useState(dayjs());
  const currentYear = date.year();
  const currentMonth = date.month();
  const isToday = num === currentDay.date()
    && currentMonth === currentDay.month() 
    && currentYear === currentDay.year();
  const isPicked = selected.includes(num);
  const style = {
    color: isToday ? 'indianred' : 'inherit',
    background: isPicked ?'green': null
  };
  const handleSelected = i => {
    //dateTime: "2020-02-28T08:30:00-08:00"

    const newdate = `${currentYear}-${currentMonth+1}-${num}`

    templateFormOpen
    ?(selected.includes(newdate)
      ? setSelected(selected.filter(day => day !== newdate))
      : setSelected(selected.concat(newdate)))
    :alert('pick a template')
  }

  return (
    <span style={style} onClick={()=> handleSelected(num)}>
      {events && events.map(e => {
        const event =
          num === dayjs(e && e.start.dateTime).date() &&
          currentMonth === dayjs(e && e.start.dateTime).month() &&
          currentYear === dayjs(e && e.start.dateTime).year()
            ? e.summary
            : null;
        return event;
      })}

      {/* {event && event.start.dateTime} */}
      {num}
    </span>
  )
}

const Calendar = ({ events, templateFormOpen, selected, setSelected, date, setDate}) => {
  // Component state
  // const [loading, setLoading] = useState(false);
  const currentYear = date.year();
  const currentMonth = date.month(); // January = 0
  const daysInMonth = date.daysInMonth();

  const firstDayOfMonth = dayjs(`${currentYear}-${currentMonth + 1}-1`);
  const weekDayOfFirstDay = firstDayOfMonth.day(); // Sunday = 0

  const lastDayOfMonth = dayjs(`${currentYear}-${currentMonth + 1}-${daysInMonth}`);
  const weekDayOfLastDay = lastDayOfMonth.day();

  const handlePrev = () => {setDate(date.subtract(1, 'month'));};

  const handleNext = () => {setDate(date.add(1, 'month'));};


  console.log(currentYear, currentMonth)
  console.log(date)
  console.log(selected)

  

  return (
    <div>
      <div className="calendar">
       {/* Header contains PrevMonth, NextMonth and Current Date */}
        <div className="header">
          <button type="button" className="nav prev" onClick={handlePrev}>
            &lt;
          </button>
          <h3 className="heading">{date.format('MMMM YYYY')}</h3>
          <button type="button" className="nav nav next" onClick={handleNext}>
            &gt;
          </button>
        </div>
        {/* Contains List of weekdays */}
        <div className="labels">
          {weekDays.map(d => (
            <span key={d}>{d}</span>
          ))}
        </div>
        {/* holds days of month*/}
        <div className="dates">
          {/* empty days from prevmonth */}
          {[...Array(weekDayOfFirstDay).keys()].map(i => (
            <span className="faded" key={i}>
              {/* {firstDayOfMonth.subtract(weekDayOfFirstDay - i, "day").date()} */}
            </span>
          ))}

          {/* labeled days for current month */}
          {[...Array(daysInMonth).keys()].map(i => {
            
            return (
              <Day 
                key={i+1}
                num={i+1}  
                events={events}
                selected={selected}
                setSelected={setSelected}
                templateFormOpen={templateFormOpen}
              />
            );
          })}
          {[...Array(6 - weekDayOfLastDay).keys()].map(i => (
            <span className="faded" key={i}>
              {/* {lastDayOfMonth.add(i + 1, "day").date()} */}
            </span>
          ))}
        </div>
      </div>
    </div>
  );
};
export default Calendar;
