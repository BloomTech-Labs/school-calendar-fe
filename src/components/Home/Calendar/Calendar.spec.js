import React from 'react';
import { shallow } from 'enzyme';
import Calendar, {button} from './Calendar';
import { create, renderer } from "react-test-renderer";

describe('Calendar', () => {
const wrapper = shallow(<Calendar />)

  it('should render correctly', () => {
    const component = shallow(<Calendar />);
    expect(Calendar).toMatchSnapshot();
  });

  it('should render days-of-the-week labels', () => {
    const strings = ['Sun', 'Mon','Tues','Wed', 'Thu','Fri', 'Sat'];
    const component = shallow(<Calendar list={strings} />);
    expect(component).toMatchSnapshot();
  });

  it('should have a btn component', ()=> {
    expect(wrapper.find('button')).toHaveLength(2);
    // expect(wrapper.find('button')
    // .toEqual('button'))
    expect('button').toBeDefined()
  });
  

});
  
