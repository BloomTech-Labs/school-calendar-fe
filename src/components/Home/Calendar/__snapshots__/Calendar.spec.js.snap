// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should render correctly 1`] = `[Function]`;

exports[`Calendar should render days-of-the-week labels 1`] = `
<div>
  <div
    className="calendar"
  >
    <div
      className="header"
    >
      <button
        className="nav prev"
        onClick={[Function]}
        type="button"
      >
        &lt;
      </button>
      <h3
        className="heading"
      >
        Feb 19 2020
      </h3>
      <button
        className="nav nav next"
        onClick={[Function]}
        type="button"
      >
        &gt;
      </button>
    </div>
    <div
      className="labels"
    >
      <span
        key="Sun"
      >
        Sun
      </span>
      <span
        key="Mon"
      >
        Mon
      </span>
      <span
        key="Tue"
      >
        Tue
      </span>
      <span
        key="Wed"
      >
        Wed
      </span>
      <span
        key="Thu"
      >
        Thu
      </span>
      <span
        key="Fri"
      >
        Fri
      </span>
      <span
        key="Sat"
      >
        Sat
      </span>
    </div>
    <div
      className="dates"
    >
      <span
        className="faded"
        key="0"
      />
      <span
        className="faded"
        key="1"
      />
      <span
        className="faded"
        key="2"
      />
      <span
        className="faded"
        key="3"
      />
      <span
        className="faded"
        key="4"
      />
      <span
        className="faded"
        key="5"
      />
      <span
        key="0"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        1
      </span>
      <span
        key="1"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        2
      </span>
      <span
        key="2"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        3
      </span>
      <span
        key="3"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        4
      </span>
      <span
        key="4"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        5
      </span>
      <span
        key="5"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        6
      </span>
      <span
        key="6"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        7
      </span>
      <span
        key="7"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        8
      </span>
      <span
        key="8"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        9
      </span>
      <span
        key="9"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        10
      </span>
      <span
        key="10"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        11
      </span>
      <span
        key="11"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        12
      </span>
      <span
        key="12"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        13
      </span>
      <span
        key="13"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        14
      </span>
      <span
        key="14"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        15
      </span>
      <span
        key="15"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        16
      </span>
      <span
        key="16"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        17
      </span>
      <span
        key="17"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        18
      </span>
      <span
        key="18"
        style={
          Object {
            "color": "indianred",
          }
        }
      >
        19
      </span>
      <span
        key="19"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        20
      </span>
      <span
        key="20"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        21
      </span>
      <span
        key="21"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        22
      </span>
      <span
        key="22"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        23
      </span>
      <span
        key="23"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        24
      </span>
      <span
        key="24"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        25
      </span>
      <span
        key="25"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        26
      </span>
      <span
        key="26"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        27
      </span>
      <span
        key="27"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        28
      </span>
      <span
        key="28"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        29
      </span>
    </div>
  </div>
</div>
`;
